addons/html_builder/static/src/plugins/card_image_option_plugin.js

```diff
@@ -0,0 +1,118 @@
+import { classAction } from "@html_builder/core/plugins/core_builder_action_plugin";
+import { Plugin } from "@html_editor/plugin";
+import { withSequence } from "@html_editor/utils/resource";
+import { registry } from "@web/core/registry";
+import { renderToElement } from "@web/core/utils/render";
+import { CardImageOption } from "./card_image_option";
+
+const ratiosOnlySupportedForTopImage = [
+    "ratio ratio-4x3",
+    "ratio ratio-16x9",
+    "ratio ratio-21x9",
+    "ratio o_card_img_ratio_custom",
+];
+const imageRelatedClasses = [
+    "o_card_img_top",
+    "o_card_img_horizontal",
+    "flex-lg-row",
+    "flex-lg-row-reverse",
+];
+const imageRelatedStyles = [
+    "--card-img-aspect-ratio",
+    "--card-img-size-h",
+    "--card-img-ratio-align",
+];
+
+class CardImageOptionPlugin extends Plugin {
+    static id = "cardImageOption";
+    static dependencies = ["remove"];
+    resources = {
+        builder_options: [
+            withSequence(30, {
+                OptionComponent: CardImageOption,
+                props: {
+                    getImageToWrapperRatio: this.getImageToWrapperRatio.bind(this),
+                },
+                selector: ".s_card",
+            }),
+        ],
+        builder_actions: {
+            setCoverImagePosition: {
+                apply: ({ editingElement, param: { mainParam: className } }) => {
+                    const imageEl = editingElement.querySelector(".o_card_img");
```

Comment:
Maybe put every thing in card_option_plugin ? (Or we have to much code and you prefer to split it ?)

Reply:
Yeah, it used to be all in the same plugin, then I split. In the original code base they're also two different option classes, and I though it made sense to group cover image-related logic in a separate plugin. But I can merge them if you prefer.